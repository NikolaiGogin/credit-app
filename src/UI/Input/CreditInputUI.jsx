import React from 'react';
import classes from './CreditInputUI.module.css';

const CalcInputUI = ({title, inscription, name, dataDirty, dataError, ...props}) => {
    return (
        <div className={classes.inputArea}>
            <label htmlFor='id' className={classes.label}>{title}</label>
            <div className={classes.creditInputWrapper}>
                <input 
                    {...props}
                    className={classes.creditInput}
                    type='text' 
                    placeholder='0'
                    autoComplete='off'
                    name={name}
                />
                <div className={classes.creditInputInscription}>
                    {inscription}
                </div>
                {(dataDirty && dataError) && <div className={classes.errorMessage}>{dataError}</div>}
            </div>
        </div>
    );
};

export default CalcInputUI;
