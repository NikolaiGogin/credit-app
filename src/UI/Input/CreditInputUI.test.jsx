import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CreditInputUI from './CreditInputUI';

const onChange = jest.fn();

describe('CreditInputUI component', () => {
    it('Сomponent renders', () => {
        render(<CreditInputUI inscription='руб.'/>);
      
        expect(screen.getByRole('textbox')).toBeInTheDocument();
        expect(screen.getByText('руб.')).toBeInTheDocument();
    });

    it('Сomponent snapshot', () => {
        const input = render(<CreditInputUI  />);

        expect(input).toMatchSnapshot();
    });

    it('onChange works', () => {
        render(<CreditInputUI onChange={onChange} />);

        userEvent.type(screen.getByRole('textbox'), '5000');

        expect(onChange).toHaveBeenCalledTimes(4);
    });
});
