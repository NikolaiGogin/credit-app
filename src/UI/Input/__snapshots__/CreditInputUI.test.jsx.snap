// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditInputUI component component snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="inputArea"
      >
        <label
          class="label"
          for="id"
        />
        <div
          class="creditInputWrapper"
        >
          <input
            autocomplete="off"
            class="creditInput"
            placeholder="0"
            type="text"
          />
          <div
            class="creditInputInscription"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="inputArea"
    >
      <label
        class="label"
        for="id"
      />
      <div
        class="creditInputWrapper"
      >
        <input
          autocomplete="off"
          class="creditInput"
          placeholder="0"
          type="text"
        />
        <div
          class="creditInputInscription"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CreditInputUI component Ð¡omponent snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="inputArea"
      >
        <label
          class="label"
          for="id"
        />
        <div
          class="creditInputWrapper"
        >
          <input
            autocomplete="off"
            class="creditInput"
            placeholder="0"
            type="text"
          />
          <div
            class="creditInputInscription"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="inputArea"
    >
      <label
        class="label"
        for="id"
      />
      <div
        class="creditInputWrapper"
      >
        <input
          autocomplete="off"
          class="creditInput"
          placeholder="0"
          type="text"
        />
        <div
          class="creditInputInscription"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
